internal/controllers/gateway/gateway_controller.go

```diff
@@ -272,6 +275,7 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	gateway := new(gatewayv1beta1.Gateway)
 	if err := r.Get(ctx, req.NamespacedName, gateway); err != nil {
 		if k8serrors.IsNotFound(err) {
+			// TODO: fill in namespace and name of gateway and delete the non-exist gateway in cache.
```

Comment:
Do you mind expanding this comment or making an issue to track this? It's not immediately obvious to me what is to be done here 🤔

Reply:
https://github.com/Kong/kubernetes-ingress-controller/issues/3069